vignettes/datatable-importing.Rmd

```diff
@@ -195,6 +195,115 @@ For more canonical documentation of defining packages dependency check the offic
+## Importing from non-r Applications {non-r-api}
 
 Some tiny parts of `data.table` C code were isolated from the R C API and can now be used from non-R applications by linking to .so / .dll files. More concrete details about this will be provided later; for now you can study the C code that was isolated from the R C API in [src/fread.c](https://github.com/Rdatatable/data.table/blob/master/src/fread.c) and [src/fwrite.c](https://github.com/Rdatatable/data.table/blob/master/src/fwrite.c).
+
+## How to convert your Depends dependency on data.table to Imports
+
+To convert a `Depends` dependency on `data.table` to an `Imports` dependency in your package, follow these steps:
+
+### Step 0.  Ensure your package is passing R CMD check initially
+
+### Step 1. Update the DESCRIPTION file to put data.table in Imports, not Depends
+
+**Before:**
+```r
+Depends:
+    R (>= 3.5.0),
+    data.table
+Imports:
+```
+
+**After:**
+```r
+Depends:
+    R (>= 3.5.0)
+Imports:
+    data.table
+```
+
+### Step 2.1: Run `R CMD check`
+
+Run `R CMD check` to identify any missing imports or symbols. This step helps:
+
+- Automatically detect any functions or symbols from `data.table` that are not explicitly imported.
+- Flag missing special symbols like `.N`, `.SD`, and `:=`.
+- Provide immediate feedback on what needs to be added to the NAMESPACE file.
+
+### Step 2.2: Modify the NAMESPACE File
+
+Based on the `R CMD check` results, ensure all used functions, special symbols, S3 generics, and S4 classes from `data.table` are imported.
+
+#### Importing Specific Functions and Symbols
+
+If you use specific functions like `data.table()`, `setDT()`, `merge()`, or special symbols like `.N`, `.SD`, `:=`, import them individually:
+```r
+importFrom(data.table, data.table, setDT, merge, .N, .SD, :=)
+```
+
+#### Importing S3 Generics
+
+If your package adds methods to any S3 generics owned by `data.table`, you need to import these generics. Check the `data.table` NAMESPACE file for the latest set of S3 generics:
+```r
+importFrom(data.table, as.data.table, groupingsets, as.IDate)
+```
+
+#### Importing S4 Classes
+
+If your package uses S4 classes from `data.table`, you need to import these classes. This is less common but necessary for advanced use cases:
+```r
+importClassesFrom(data.table, data.table, IDate, ITime)
+```
+
+#### Blanket Import
+
+Alternatively, you can import all functions from `data.table` at once, though this is generally not recommended:
+
+```r
+import(data.table)
+```
+
+**Justification for Avoiding Blanket Imports:**
+1. **Documentation**: The NAMESPACE file can serve as good documentation of how you depend on certain packages.
+2. **Avoiding Conflicts**: Blanket imports leave you open to subtle breakage. For example, if you import(pkgA) and import(pkgB), but later pkgB exports a function also exported by pkgA, this will break your package due to conflicts in your namespace, which is disallowed by `R CMD check` and CRAN.
```

Comment:
```suggestion
2. **Avoiding Conflicts**: Blanket imports leave you open to subtle breakage. For example, if you `import(pkgA)` and `import(pkgB)`, but later pkgB exports a function also exported by pkgA, this will break your package due to conflicts in your namespace, which is disallowed by `R CMD check` and CRAN.
```